lib/server_sent_event_stage.ex

```diff
@@ -22,23 +22,25 @@ defmodule ServerSentEventStage do
   """
   def start_link(args) do
     url = Keyword.fetch!(args, :url)
+    headers = Keyword.get(args, :headers, [])
     opts = Keyword.take(args, ~w(debug name timeout spawn_opt)a)
-    GenStage.start_link(__MODULE__, url, opts)
+    GenStage.start_link(__MODULE__, %{url: url, headers: headers}, opts)
```

Comment:
We're `fetch!`ing the URL more to ensure that it's there. It'll be simpler to pass the `args` directly into `init`:
```elixir
_url = Keyword.fetch!(args, :url)
opts = ...
GenStage.start_link(__MODULE__, args, opts)
```